### DISCUSSION TITLE
Documentation: wsl -d doesn't give me a shell

### DISCUSSION THREAD

**Discussion by kagehisa** [nix-community/NixOS-WSL#471] [Q&A] [answered]
I followed the documentation to install the nixos-wsl image. I deviated from it by using 
```
wsl --import NixOS %USERPROFILE%\NixOS\ nixos-wsl.tar.gz --version 2
```
instead of the documented import command without the version. 
The import works fine, but `wsl -d` just drops me back in to the powershell and `wsl -l -v` tells me NixOS is runnning.
After a bit of snooping around I found tried the following:
```
wsl -d NixOS  --user root -e /nix/var/nix/profiles/system/activate
```
This runs the activation script and drops me into a shell, but after I ext this shell and try to get back into the system with `wsl -d NixOS` I get the same behavior as before. The commend gets executed and I am back at the powershell without any output from wsl or NixOS.

Is there anything I'm missing?

**Comment by nzbr** [comment]
What does `wsl --version` show for you?

**Reply by kagehisa** [reply]
Hi, `wsl --version` is not available for my wsl installation as it seems.
But `wsl --status` gives me:
```
Standard Distribution: Ubuntu-24.04
Standardversion: 2
Kernel-Version: 5.10.16
```

WSL is running at version 2. Am I missing something?

**Reply by 573** [reply, marked as answer]
Your installed `wsl` seems not recent enough. In very recent `wsl`s (https://github.com/microsoft/WSL/releases/tag/2.2.3) though the command might be `wslinfo --wsl-version`

**Reply by kagehisa** [reply]
Ok, that was the Problem. It took soome tweaking but I was able to install WSL Release 2.2.4. 
I now get a shell. 

Thanks for the help!
